packages/stark-ui/src/modules/route-search/components/route-search.component.ts

```diff
@@ -0,0 +1,287 @@
+		if (!this.hide) {
+			// set focus on the search field automatically
+			setTimeout(() => {
+				const inputField: HTMLElement = this.elementRef.nativeElement.querySelector(".search-field-input");
+				inputField.focus();
+			}, 80);
+		}
+	}
+
+	/**
+	 * @ignore
+	 */
+	public trackPath(_index: number, path: RouteEntry): string {
+		return path.targetState;
+	}
+
+	/**
+	 * Retrieve the list of routes from a {@link StarkMenuConfig} object which the user have passed to the component
+	 */
+	public constructRouteEntriesFromMenuConfig(menuConfig: StarkMenuConfig): RouteEntry[] {
+		let routesToDisplay: RouteEntry[] = [];
+
+		if (menuConfig !== undefined && menuConfig.menuSections !== undefined) {
+			for (const section of menuConfig.menuSections) {
+				for (const group of section.menuGroups) {
+					routesToDisplay = [...routesToDisplay, ...this.extractRoutesFromMenuGroup(group)];
+				}
+			}
+		}
+
+		if (menuConfig !== undefined && menuConfig.menuGroups !== undefined) {
+			for (const group of menuConfig.menuGroups) {
+				routesToDisplay = [...routesToDisplay, ...this.extractRoutesFromMenuGroup(group)];
+			}
+		}
+
+		return routesToDisplay;
+	}
+
+	/**
+	 * Recursive method to extract routes from a StarkMenuGroup object.
+	 * @param group - the group which entries need to be extracted
+	 * @returns the extracted array of {@link RouteEntry} objects
+	 */
+	public extractRoutesFromMenuGroup(group: StarkMenuGroup): RouteEntry[] {
+		let routesToDisplay: RouteEntry[] = [];
+
+		if (group.isVisible && group.isEnabled && group.targetState) {
+			routesToDisplay.push({
+				label: group.label,
+				targetState: group.targetState,
+				targetStateParams: group.targetStateParams
+			});
+		}
+		if (group.entries) {
+			for (const subGroup of group.entries) {
+				routesToDisplay = [...routesToDisplay, ...this.extractRoutesFromMenuGroup(subGroup)];
+			}
+		}
+
+		return routesToDisplay;
+	}
+
+	/**
+	 * Retrieve the list of routes from the Router if the user haven't passed a list to the component
+	 */
+	public constructRouteEntriesFromRouterStates(): RouteEntry[] {
+		const routesToDisplay: RouteEntry[] = [];
+		for (const state of this.starkRoutingService.getStatesConfig()) {
+			const ng2State: Ng2StateDeclaration = state;
+			const regexInitExitStateName: RegExp = new RegExp("(" + starkAppInitStateName + "|" + starkAppExitStateName + ")");
+			if (
+				ng2State.name !== undefined &&
+				ng2State.url !== undefined &&
+				ng2State.url.length > 1 &&
+				!ng2State.abstract &&
+				!ng2State.loadChildren &&
+				!ng2State.name.match(regexInitExitStateName)
+			) {
+				const translationKey: string = this.starkRoutingService.getTranslationKeyFromState(ng2State.name);
+				routesToDisplay.push({ label: translationKey, targetState: ng2State.name });
+			}
+		}
+
+		return routesToDisplay;
+	}
+
+	/**
+	 * Translate the label of all elements in the {@link RouteEntry} object.
+	 * This is useful as the language can change during runtime.
```

Comment:
If we simply put the translation in the HTML like this:

```html
<mat-option
	(onSelectionChange)="redirect(option)"
	*ngFor="let option of (filteredRouteEntries | async); trackBy: trackPath"
	[value]="option.label  | translate"
>
	<p>{{ option.label | translate }}</p>
</mat-option>
```

That means it won't translated if the language changes during runtime ?

Reply:
In fact, the translation is not the problem. The problem is the sorting because they options should be sorted using the translated values so it cannot be done only in the template cause there we only have the translation keys ;)